linkml/generators/sqltablegen.py

```diff
@@ -220,13 +225,19 @@ def strip_newlines(txt: str | None) -> str:
                         fk = sql_name(self.get_id_or_key(s.range, sv))
                         args = [ForeignKey(fk)]
                     field_type = self.get_sql_range(s, schema)
+                    fk_index_cond = (s.key or s.identifier) and self.autogenerate_fk_index
+                    pk_index_cond = is_pk and self.autogenerate_pk_index
+                    is_index = fk_index_cond or fk_index_cond
                     col = Column(
                         sql_name(sn),
                         field_type,
                         *args,
                         primary_key=is_pk,
                         nullable=not s.required,
+                        index=(fk_index_cond or pk_index_cond),
```

Comment:
you can use the variable `is_index` (currently incorrect, but will be OK once the typo is fixed) here